### PR TITLE
[scanning] Implement ReviewScreen and ConfirmationScreen (screens/scanning)

### PR DESCRIPTION
**Pull request by selene-huang**
[//]: # "These comments are meant for your reference. They are invisible and don't need to be deleted!"

### Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."
- Updated ScanningContext to include edit, delete, and reset functionalities (EDIT_VOUCHER, DELETE_VOUCHER, and NEW_INVOICE respectively)
- Implemented hifi interfaces for Review screen and Confirmation screen
- Moved some frontend components into common utilities
- Fixed issue in ConfirmValueScreen that led to values being saved in dollars rather than cents
- Input validation now fully implemented on the Manual Voucher Add screen

**Notes on the Review Screen:**

We had initially intended there to be a set box for the vouchers to be viewed, leaving the submit button in the same position no matter what and displaying up to 4 or 5 vouchers at a time. The user was then supposed to scroll through the rest. However, we found there was no easy way to indicate that the user should scroll, so we opted instead to include the submit button as part of the scroll.

I have set a minimum height for the vouchers being displayed so that the submit button at least has some consistency in its positioning.

### Screenshots
[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy!"
Serial Number Validation: 
image

Review Screen (short):
image

Review Screen (with scrolling):
image

Android Review Screen Alert Samples:
image
image

Confirmation Screen:
image

## How to review
[//]: # "Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?"

Run `npx expo start`. Navigate the scanning flow and press Add Manually. Input a serial number in the range from 100-110 (but note that some may be invalid because they are already in use). Check that input validation works as intended, and that you can edit and delete vouchers as you please before submitting. Check that the confirmation screen navigates to the correct screens afterwards.

### Online sources
[//]: # "Optional - copy links to any tutorials or documentation that was useful to you when working on this PR"

React native does not support inputs for alerts on Android. The [Dialog package](https://www.npmjs.com/package/react-native-dialog?activeTab=readme) is a useful work around.

### Related PRs
[//]: # "Optional - related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

Refactors code from #29 and #58 
Builds upon input validation from #49 

[//]: # "This tags the project leader as a default. Feel free to change, or add on anyone who you should be in on the conversation."
🥦 CC: @sauhardjain

### CODE REVIEW

File: src/components/InputField/InputField.tsx [hunk 1]
```diff
@@ -26,11 +28,20 @@ export default function InputField({
...
-        validate?.(value);
+        if (validate) {
+          validate(value);
+        }
       }}
       onFocus={() => setIsActive(true)}
       onChangeText={onChange}
-      style={isActive ? Styles.FormFieldFocus : Styles.FormField}
+      style={
+        // eslint-disable-next-line no-nested-ternary
```
Comment (kbrgl) [inline src/components/InputField/InputField.tsx]: I suggest rewriting as an if statement—not the most readable right now. Here's some code you can use if you want. Add it before the return:
```
let textInputStyle;

if (active) {
  textInputStyle = Styles.FormFieldFocus;
} else if (isValid) {
  textInputStyle = Styles.FormField;
} else {
  textInputStyle = Styles.FormFieldError;
}
```
and then pass `textInputStyle` in here.

File: src/components/common/BackButton.tsx
Comment (kbrgl) [inline src/components/common/BackButton.tsx:1]: Pretty sure that implementing a custom button here is not the move, regardless of what the Figma says. React Navigation should IIRC already allow you to add a header with a back button. Further, I believe React Navigation also allows you to customize this header: https://reactnavigation.org/docs/header-buttons/#overriding-the-back-button

cc @sauhardjain

File: src/navigation/stacks/ScannerStackNavigator.tsx [hunk 2]
```diff
@@ -28,6 +31,15 @@ export default function ScannerStackNavigator() {
           name="ConfirmValueScreen"
           component={ConfirmValueScreen}
         />
+        <ScannerStack.Screen name="ReviewScreen" component={ReviewScreen} />
+        <ScannerStack.Screen
+          name="ConfirmationScreen"
+          component={ConfirmationScreen}
+        />
+        <ScannerStack.Screen
+          name="Transactions"
```
Comment (kbrgl) [inline src/navigation/stacks/ScannerStackNavigator.tsx]: Rename to TransactionsScreen?

File: src/screens/auth/ForgotPasswordScreen.tsx [hunk 3]
```diff
@@ -48,14 +45,8 @@ export default function ForgotPasswordScreen({
...
-      <StandardHeader alignment="flex-start">
-        <BackButtonContainer>
-          <RootNavBackButton onPress={() => navigation.navigate('Login')}>
-            <DarkGrayText>
-              <Icon name="left" size={14} color={Colors.darkGray} /> Back
-            </DarkGrayText>
-          </RootNavBackButton>
-        </BackButtonContainer>
+      <StandardHeader>
+        {BackButton(() => navigation.navigate('Login'))}
```
Comment (kbrgl) [inline src/screens/auth/ForgotPasswordScreen.tsx]: This is not how you use a React component—it should always be used with the JSX syntax (`<BackButton onPress={...} />`. Here, you're just calling the component as a function. Resources on why not to do this:

https://dev.to/igor_bykov/react-calling-functional-components-as-functions-1d3l
https://kentcdodds.com/blog/dont-call-a-react-function-component

File: src/screens/auth/LoginScreen.tsx [hunk 4]
```diff
@@ -67,13 +64,7 @@ export default function LoginScreen({
...
     <LoginContainer>
       <StandardHeader alignment="flex-start">
-        <BackButtonContainer>
-          <RootNavBackButton onPress={() => navigation.navigate('Start')}>
-            <DarkGrayText>
-              <Icon name="left" size={14} color={Colors.darkGray} /> Back
-            </DarkGrayText>
-          </RootNavBackButton>
-        </BackButtonContainer>
+        {BackButton(() => navigation.navigate('Start'))}
```
Comment (kbrgl) [inline src/screens/auth/LoginScreen.tsx]: Don't call here

File: src/screens/scanning/ConfirmationScreen.tsx [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+  const { count } = route.params;
+  const { dispatch } = useScanningContext();
+
+  useEffect(() => {
+    newInvoice(dispatch);
+  }, [dispatch]);
+
+  return (
+    <SafeArea>
+      <StandardHeader>{BackButton(() => navigation.goBack())}</StandardHeader>
```
Comment (kbrgl) [inline src/screens/scanning/ConfirmationScreen.tsx]: Don't call here

Comment (kbrgl) [reply src/screens/scanning/ConfirmationScreen.tsx]: (To reiterate, use as a React component with JSX syntax)

File: src/screens/transactions/TransactionDetailsScreen.tsx [hunk 6]
```diff
@@ -66,15 +61,7 @@ export default function TransactionDetailsScreen({
...
-        <BackButtonContainer>
-          <RootNavBackButton
-            onPress={() => navigation.navigate('TransactionsScreen')}
-          >
-            <DarkGrayText>
-              <Icon name="left" size={14} color={Colors.darkGray} /> Back
-            </DarkGrayText>
-          </RootNavBackButton>
-        </BackButtonContainer>
+        {BackButton(() => navigation.navigate('TransactionsScreen'))}
```
Comment (kbrgl) [inline src/screens/transactions/TransactionDetailsScreen.tsx]: Don't call here

**Review by kbrgl — CHANGES_REQUESTED** [review]
Great PR! Love the commit discipline. :)

A few minor changes here and there. Slack me if you have any questions.

**Review by sauhardjain — COMMENTED** [review]
Leaving notes for bugs to be fixed:
1. Async validation calls on voucher entry form taking too long: results in an error message at first which goes away on a second try.
2. Screen shifts down when keyboard opens.

**Review by sauhardjain — APPROVED** [review]
lgtm THANKS @selene-huang
